pyproject.toml

```diff
@@ -12,20 +12,16 @@ readme = "README.md"
 requires-python = ">=3.8"
 license = {text = "Apache 2.0"}
 dependencies = [
-    "h5py>=3.7.0",
-    "mpi4py>=3.1.4",
-    "netcdf4>=1.6.3",
-    "certifi>=2023.7.22",
-    "numpy>=1.22.4",
-    "nvidia_dali_cuda110>=1.16.0",
-    "pytest>=6.0.0",
-    "ruamel.yaml>=0.17.22",
-    "setuptools>=67.6.0",
     "torch>=1.12",
```

Comment:
I am going to be a broken record here but I don't believe this version for `torch` is accurate.

Reply:
Good point @dallasfoster , I think this needs to be torch 2.0. I will do some local testing and then update it accordingly

Reply:
Added this. Tested things in a blank conda environment and the below sequence of commands work:

```
conda create --name test-env python=3.8
conda activate test-env
pip install .
pip install .[dev]
cd test/
pytest
```
cc @NickGeneva

Reply:
Can we see `conda list`, for completeness?

Reply:
Sure! This is what my env shows:

```
conda list
# packages in environment at /raid/kaustubh_backup/anaconda3/envs/modulus-dep-cleanup-test01:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
aiobotocore               2.5.4                    pypi_0    pypi
aiohttp                   3.8.5                    pypi_0    pypi
aioitertools              0.11.0                   pypi_0    pypi
aiosignal                 1.3.1                    pypi_0    pypi
asciitree                 0.3.3                    pypi_0    pypi
astunparse                1.6.3                    pypi_0    pypi
async-timeout             4.0.3                    pypi_0    pypi
attrs                     23.1.0                   pypi_0    pypi
black                     22.10.0                  pypi_0    pypi
botocore                  1.31.17                  pypi_0    pypi
bzip2                     1.0.8                h7f98852_4    conda-forge
ca-certificates           2023.7.22            hbcca054_0    conda-forge
certifi                   2023.7.22                pypi_0    pypi
charset-normalizer        3.2.0                    pypi_0    pypi
cmake                     3.27.5                   pypi_0    pypi
colorama                  0.4.6                    pypi_0    pypi
coverage                  6.5.0                    pypi_0    pypi
dm-tree                   0.1.8                    pypi_0    pypi
entrypoints               0.4                      pypi_0    pypi
exceptiongroup            1.1.3                    pypi_0    pypi
fasteners                 0.19                     pypi_0    pypi
filelock                  3.12.4                   pypi_0    pypi
frozenlist                1.4.0                    pypi_0    pypi
fsspec                    2023.9.1                 pypi_0    pypi
gast                      0.5.4                    pypi_0    pypi
h5py                      3.9.0                    pypi_0    pypi
idna                      3.4                      pypi_0    pypi
iniconfig                 2.0.0                    pypi_0    pypi
interrogate               1.5.0                    pypi_0    pypi
jinja2                    3.1.2                    pypi_0    pypi
jmespath                  1.0.1                    pypi_0    pypi
ld_impl_linux-64          2.40                 h41732ed_0    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 13.2.0               h807b86a_1    conda-forge
libgomp                   13.2.0               h807b86a_1    conda-forge
libnsl                    2.0.0                h7f98852_0    conda-forge
libsqlite                 3.43.0               h2797004_0    conda-forge
libuuid                   2.38.1               h0b41bf4_0    conda-forge
libzlib                   1.2.13               hd590300_5    conda-forge
lit                       16.0.6                   pypi_0    pypi
markupsafe                2.1.3                    pypi_0    pypi
mpmath                    1.3.0                    pypi_0    pypi
multidict                 6.0.4                    pypi_0    pypi
mypy-extensions           1.0.0                    pypi_0    pypi
ncurses                   6.4                  hcb278e6_0    conda-forge
networkx                  3.1                      pypi_0    pypi
numcodecs                 0.11.0                   pypi_0    pypi
numpy                     1.24.4                   pypi_0    pypi
nvidia-cublas-cu11        11.10.3.66               pypi_0    pypi
nvidia-cuda-cupti-cu11    11.7.101                 pypi_0    pypi
nvidia-cuda-nvrtc-cu11    11.7.99                  pypi_0    pypi
nvidia-cuda-runtime-cu11  11.7.99                  pypi_0    pypi
nvidia-cudnn-cu11         8.5.0.96                 pypi_0    pypi
nvidia-cufft-cu11         10.9.0.58                pypi_0    pypi
nvidia-curand-cu11        10.2.10.91               pypi_0    pypi
nvidia-cusolver-cu11      11.4.0.1                 pypi_0    pypi
nvidia-cusparse-cu11      11.7.4.91                pypi_0    pypi
nvidia-dali-cuda110       1.29.0                   pypi_0    pypi
nvidia-modulus            0.3.0a0                  pypi_0    pypi
nvidia-nccl-cu11          2.14.3                   pypi_0    pypi
nvidia-nvtx-cu11          11.7.91                  pypi_0    pypi
onnx                      1.14.1                   pypi_0    pypi
openssl                   3.1.2                hd590300_0    conda-forge
packaging                 23.1                     pypi_0    pypi
pandas                    2.0.3                    pypi_0    pypi
pathspec                  0.11.2                   pypi_0    pypi
pip                       23.2.1             pyhd8ed1ab_0    conda-forge
pluggy                    1.3.0                    pypi_0    pypi
protobuf                  4.24.3                   pypi_0    pypi
py                        1.11.0                   pypi_0    pypi
pytest                    7.4.2                    pypi_0    pypi
python                    3.8.17          he550d4f_0_cpython    conda-forge
python-dateutil           2.8.2                    pypi_0    pypi
pytz                      2023.3.post1             pypi_0    pypi
pyyaml                    6.0.1                    pypi_0    pypi
readline                  8.2                  h8228510_1    conda-forge
s3fs                      2023.9.1                 pypi_0    pypi
setuptools                68.2.2             pyhd8ed1ab_0    conda-forge
six                       1.16.0                   pypi_0    pypi
sympy                     1.12                     pypi_0    pypi
tk                        8.6.12               h27826a3_0    conda-forge
toml                      0.10.2                   pypi_0    pypi
tomli                     2.0.1                    pypi_0    pypi
torch                     2.0.1                    pypi_0    pypi
triton                    2.0.0                    pypi_0    pypi
typing-extensions         4.8.0                    pypi_0    pypi
tzdata                    2023.3                   pypi_0    pypi
urllib3                   1.26.16                  pypi_0    pypi
wheel                     0.41.2             pyhd8ed1ab_0    conda-forge
wrapt                     1.15.0                   pypi_0    pypi
xarray                    2023.1.0                 pypi_0    pypi
xz                        5.2.6                h166bdaf_0    conda-forge
yarl                      1.9.2                    pypi_0    pypi
zarr                      2.16.1                   pypi_0    pypi
```